### PR TITLE
Closes #696 Elide 7 + Spring boot 3

### LINKED ISSUE #696: Elide 7 + Spring boot 3

### PR DESCRIPTION
**Pull request by bukajsytlos**
Closes #696 Elide 7 + Spring boot 3

### CODE REVIEW

**Review by Sheikah45 — DISMISSED** [review]

File: src/main/java/com/faforever/api/user/GogService.java [hunk 1]
```diff
@@ -108,11 +108,14 @@ private GogGamesListPage getGamesListPage(String gogUsername, int page) {
...
-        default -> e;
-      };
+      switch ((HttpStatus) e.getStatusCode()) {
+        case FORBIDDEN:
+          throw ApiException.of(ErrorCode.GOG_LINK_GAMES_NOT_PUBLIC);
+        case NOT_FOUND:
+          throw ApiException.of(ErrorCode.GOG_LINK_USER_NOT_FOUND);
+        default:
+          throw e;
+      }
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/user/GogService.java:118]: Please use new switch expressions with `->`

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/api/user/GogService.java (hunk 1 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/user/GogService.java:118]: it's not working for me. feel free to try. it's sealed interface now and looks like it isn't that much supported with java 17. If I've enabled preview features, it does not complain (gradle), but idea refuses to compile

File: src/main/java/com/faforever/api/challonge/ChallongeController.java [hunk 2]
```diff
@@ -75,12 +75,12 @@ public CompletableFuture<ResponseEntity<String>> write(@RequestBody(required = f
...
 
   @RequiredArgsConstructor
-  private static class ChallongeUriTemplateHandler extends DefaultUriTemplateHandler {
+  private static class ChallongeUriTemplateHandler extends DefaultUriBuilderFactory {
     private final String apiKey;
 
     @Override
-    protected UriComponentsBuilder initUriComponentsBuilder(String uriTemplate) {
-      return super.initUriComponentsBuilder(uriTemplate).queryParam("api_key", apiKey);
+    public UriBuilder uriString(String uriTemplate) {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/challonge/ChallongeController.java:82]: not sure about this implementation. if anybody know how to test pls test or tell me :)

Comment (bukajsytlos) [reply src/main/java/com/faforever/api/challonge/ChallongeController.java:82]: looks like it is working

File: src/main/java/com/faforever/api/user/GogService.java (hunk 1 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/user/GogService.java:118]: Ah ok. Well then for the time bein its fine.

**Pull request merged**